src/runtime/vdom/vdom-render.ts

```diff
@@ -757,32 +780,42 @@ const relocateSlotContent = (elm: d.RenderNode) => {
       }
     }
 
+    // if we're dealing with any type of element (capable of itself being a
+    // slot reference or containing one) then we recur
     if (childNode.nodeType === NODE_TYPE.ElementNode) {
       relocateSlotContent(childNode);
     }
   }
 };
 
-const isNodeLocatedInSlot = (nodeToRelocate: d.RenderNode, slotNameAttr: string) => {
+/**
+ * Check whether a node is located in a given named slot.
+ *
+ * @param nodeToRelocate the node of interest
+ * @param slotName the slot name to check
+ * @returns whether the node is located in the slot or not
+ */
+const isNodeLocatedInSlot = (nodeToRelocate: d.RenderNode, slotName: string): boolean => {
   if (nodeToRelocate.nodeType === NODE_TYPE.ElementNode) {
-    if (nodeToRelocate.getAttribute('slot') === null && slotNameAttr === '') {
+    if (nodeToRelocate.getAttribute('slot') === null && slotName === '') {
+      // TODO why is this true here?
```

Comment:
Minor - unaddressed TODO. Can we either put this in the backlog or address this before it lands?

Reply:
yeah, this one is a bit confusing to me, it _looks_ like the code basically assumes that if a slot is unnamed then any node that we check will end up being marked as part of the slot, which is a bit confusing and maybe not right? I'll just explain that for now I think